builtin/providers/aws/resource_aws_iam_server_certificate_test.go

```diff
@@ -51,6 +51,41 @@ func TestAccAWSIAMServerCertificate_name_prefix(t *testing.T) {
 	})
 }
 
+func TestAccAWSIAMServerCertificate_recreate(t *testing.T) {
+	var cert iam.ServerCertificate
+
+	testDestroyCert := func(*terraform.State) error {
+		// reach out and DELETE the Cert
+		conn := testAccProvider.Meta().(*AWSClient).iamconn
+		_, err := conn.DeleteServerCertificate(&iam.DeleteServerCertificateInput{
+			ServerCertificateName: cert.ServerCertificateMetadata.ServerCertificateName,
+		})
+
+		if err != nil {
+			return fmt.Errorf("Error destorying cert in test: %s", err)
+		}
+
+		return nil
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckIAMServerCertificateDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccIAMServerCertConfig_random,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCertExists("aws_iam_server_certificate.test_cert", &cert),
+					testAccCheckAWSServerCertAttributes(&cert),
+					testDestroyCert,
+				),
+				ExpectNonEmptyPlan: true,
+			},
```

Comment:
In other `_disappears` tests I add a second step w/ a blank config and expect an empty plan, which is a nice extra check that the state is actually removed.